frontend/cs/scs/builder.go

```diff
@@ -686,3 +687,57 @@ func (builder *builder) ToCanonicalVariable(v frontend.Variable) frontend.Canoni
 		return term
 	}
 }
+
+// GetWireGates returns the pairs (gateId, wireLocation) for the given wires in
+// the compiled constraint system:
+//   - gateId is the index of the gate in the constraint system.
+//   - wireLocation is the location of the wire in the gate (0=xA or 1=xB).
+//
+// If the argument addMissing is true, then the function will add a new
+// instruction for each wire that is not found in the constraint system. This
+// may happen when getting the gate for a witness which is not used in gates.
+// Otherwise, when addMissing is false, the function returns an error if a wire
+// is not found in the constraint system.
+//
+// The order of the returned pairs is not the same as for the given arguments.
+func (builder *builder) GetWireGates(wires []frontend.Variable, addMissing bool) ([][2]int, error) {
```

Comment:
noticed that if we have duplicate wireIDs in the inputs, then the length of the return variable will remove duplicates; might be worth documenting, don't know about current caller usage

Reply:
Yes, it would be better to document it

Reply:
Documented.